azurelinuxagent/agent.py

```diff
@@ -157,6 +162,8 @@ def main(args=[]):
                 agent.daemon()
             elif command == "run-exthandlers":
                 agent.run_exthandlers()
+            elif command == "show-configuration":
```

Comment:
@boumenot It dumps the configuration state as the agent "sees" it. So, for unspecified values, it will display their default settings. I believe that is the intent of the request.